### PR TITLE
feat: gravity bridge WeightedOperations implementation

### PR DESCRIPTION
**Pull request by adamewozniak**
## Description

Gravity bridge sim implementation 

Relates to #389

----

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added appropriate labels to the PR
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/gravity/module.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// type check to ensure the interface is properly implemented
+var (
+	_ module.AppModuleSimulation = AppModule{}
+)
+
+// AppModuleBasic object for module implementation
+type AppModuleBasic struct{}
```
Comment (alexanderbez) [inline x/gravity/module.go:34]: We shouldn't have to redefine`AppModuleBasic`. We can just embed the original one defined by the gravity module and override and method(s) that we need to.

File: x/gravity/module.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	AppModuleBasic
+	keeper        keeper.Keeper
+	bankKeeper    bankkeeper.Keeper
+	accountKeeper distrtypes.AccountKeeper
+	stakingKeeper stakingkeeper.Keeper
+	cdc           codec.Codec
+}
+
+// NewAppModule creates a new AppModule Object
+func NewAppModule(
```
Comment (alexanderbez) [inline x/gravity/module.go:92]: Same for `AppModule`. We don't need to redefine it. Just embed the original and override any method(s) we need to.

**Review by alexanderbez — COMMENTED** [review]
Nice! So does this actually fix the issue? Or do we still see sim failures?

**Comment by adamewozniak** [comment]
> Nice! So does this actually fix the issue? Or do we still see sim failures?

This fixes half the problem, the missing eth / orch addresses. The [other issue](https://github.com/Gravity-Bridge/Gravity-Bridge/issues/35) is open in the gravity bridge repo. So, more sims should work, but not all of them

**Comment by alexanderbez** [comment]
I've disabled multi-seed sims in `main`. Let's re-enable them here to see which ones fail?

**Comment by alexanderbez** [comment]
@adamewozniak lmk when you want me to take another peek at this 👍

**Comment by adamewozniak** [comment]
> @adamewozniak lmk when you want me to take another peek at this 👍

Should be ready except for taking multi-seed tests back out 

Btw, waiting on reviews / workflows for [the gravity bridge pr](https://github.com/Gravity-Bridge/Gravity-Bridge/pull/36)

**Comment by alexanderbez** [comment]
@adamewozniak these changes don't fix multi-seeds, but I guess that's expected? But more importantly, it also breaks the non-determinism sim, which **should** pass. Could you look into that?

**Comment by adamewozniak** [comment]
> @adamewozniak these changes don't fix multi-seeds, but I guess that's expected? But more importantly, it also breaks the non-determinism sim, which **should** pass. Could you look into that?

Definitely has to do with `generateEthAddress` - I'll find a way to create these deterministically

**Comment by alexanderbez** [comment]
@adamewozniak there is a variant of `generateEthAddress` that accepts a `rand.Read`.

**Comment by toteki** [comment]
Is this one ready to review with intent to merge or is it still in progress?

**Comment by adamewozniak** [comment]
> Is this one ready to review with intent to merge or is it still in progress?

Still in progress, I'll move it back to a draft for the time being

**Comment by adamewozniak** [comment]
closing this until I find some way to make the tests more efficient

**Pull request closed without merging**